interface/procedure_tools/labcorp/ereq_form.php

```diff
@@ -0,0 +1,780 @@
+        $pdfContent .= '<tr><td class="width50" style="width:50%;">Patient Signature</td></tr>';
+        $pdfContent .= '</table>';
+        $pdfContent .= '</td>';
+        $pdfContent .= '<td class="width50">';
+        $pdfContent .= '<table style="width:150px;">';
+        $pdfContent .= '<tr>';
+        $pdfContent .= '<td class="width50" style="width:50%;height: 40px;padding-top:40px;border-bottom:1px solid #000;">&nbsp;</td>';
+        $pdfContent .= '</tr>';
+        $pdfContent .= '<tr><td class="width50" style="width:50%;">Date</td></tr>';
+        $pdfContent .= '</table>';
+        $pdfContent .= '</td>';
+        $pdfContent .= '</tr>';
+        $pdfContent .= '</table>';
+        $pdfContent .= '</td>';
+        $pdfContent .= '</tr>';
+        $pdfContent .= '</tbody>';
+        $pdfContent .= '</table>';
+        $pdfContent .= '</div>';
+        $pdfContent .= '</body>';
+        $pdfContent .= '</html>';
+        $mpdf->WriteHTML($styleSheet, 1);
+        $mpdf->WriteHTML($pdfContent, 2);
+
+        $directory = $GLOBALS["OE_SITE_DIR"] . "/documents/ereq/" . $pid . "/";
+        if (!is_dir($directory)) {
+            if (!mkdir($directory, 0755, true) && !is_dir($directory)) {
+                throw new \RuntimeException(sprintf('Directory "%s" was not created', $directory));
+            }
+        }
+        $unique = date('y-m-d-His', time());
+        $filename = "ereq_" . $unique . "_order_" . $form_id . ".pdf";
+//-----------patch in exclude dodoc
+        if ($_REQUEST['debug'] || !$doDoc) {
+            $mpdf->Output($directory . $filename, "I");
+            return;
+        }
+
+        $mpdf->Output($directory . $filename, "F");
+
+// register the new document
+        $category = sqlQuery("SELECT id FROM categories WHERE name LIKE ?", array("LabCorp"));
+        if (!$category['id']) {
+            $category = sqlQuery("SELECT id FROM categories WHERE name LIKE ?", array('Lab Report'));
+        }
+        $DOCUMENT_CATEGORY = $category['id'];
+
+        $d = new Document();
+        $d->name = $filename;
+        $d->storagemethod = 0;
+        $d->url = "file://" . $directory . $filename;
+        $d->mimetype = "application/pdf";
+        $d->size = filesize($directory . $filename);
+        $d->owner = $_SESSION['authUserID'];
+        $d->hash = sha1_file($directory . $filename);
+        $d->type = $d->type_array['file_url'];
+        $d->list_id = (int)$form_id;
+        $d->set_foreign_id($pid);
+        $d->persist();
+        $d->populate();
+        $docid = $d->get_id();
+        $unique = date('y-m-d-H:i:s', time());
+        $documentationOf = "$unique";
+        sqlStatement(
+            "UPDATE documents SET documentationOf = ? WHERE id = ?",
+            array($documentationOf, $docid)
+        );
+        // update cross reference
+        $query = "REPLACE INTO categories_to_documents set category_id = ?, document_id = ?";
+        sqlStatement($query, array($DOCUMENT_CATEGORY, $docid));
```

Comment:
use one of the standard functions for creating a document rather than doing it by scratch (or else things will get missed such as relative pathing and encryption; plus it's easier since just a one or two liner)